HW-6 LiveData 적용

BeokBeok opened the pull request:
Live Data 및 View Pager 적용 하였습니다.

BeokBeok committed:
HW-6 ObservableField -> LiveData 변경

BeokBeok committed:
HW-6 BaseRecyclerView 적용

- 참고 소스
https://github.com/sjjeong/GoogryBaseRecyclerView.git

BeokBeok committed:
HW-6 ViewModel AAC 제거

BeokBeok committed:
HW-6 ActionBar 제거

BeokBeok committed:
HW-6 ActionBar 제거

BeokBeok committed:
HW-6 Text 크기 및 마진 설정

BeokBeok committed:
HW-6 View Pager 적용

- UpbitFragment -> UpbitOutlineFragment 로 변경
- ProgressBar, Selected Text 관련 로직 제거

BeokBeok committed:
HW-6 안드로이드 스튜디오 3.5 버전에서 구동되도록 변경

BeokBeok committed:
HW-6 targetSDKVersion 28 -> 29 변경 및 변경으로 인한 에러 수정

BeokBeok committed:
HW-6 코인 관련 LiveData 에 postValue 로 세팅하도록 변경

NackUn commented on BeokBeok/app/src/main/java/com/architecturestudy/base/BaseRecyclerView.kt:
```diff
@@ -0,0 +1,73 @@
+        override fun onCreateViewHolder(
+            parent: ViewGroup,
+            viewType: Int
+        ): BaseViewHolder<VDB> =
```
: BaseViewHolder<VDB> 이부분 생략가능하지 않을까요?

BeokBeok replied:
onCreateViewHolder 를 오버라이딩하여 그대로 사용하였기 때문에
생략을 하지 않아도 될 것 같습니다.

NackUn replied:
넵 고생하셨습니다!

NackUn commented on BeokBeok/app/src/main/java/com/architecturestudy/base/BaseRecyclerView.kt:
```diff
@@ -0,0 +1,73 @@
+    ) {
+        private val binding: VDB = DataBindingUtil.bind(itemView)!!
+
+        fun onBind(item: Any?) {
```
석준님 예제에서는 onBindViewHolder로 되어있는데 onBind와의 차이가 있을까요? item을 Bind하느냐, ViewHolder를 Bind하느냐 차이일까요?

BeokBeok replied:
차이는 없습니다.
다만, 저에게는 onBind 함수명이 더 간결한 것 같아서 이렇게 명명하였습니다.

NackUn replied:
아 그렇군요! 감사합니다 ㅎㅎ

NackUn commented on BeokBeok/app/src/main/java/com/architecturestudy/upbitmarket/UpbitContentsFragment.kt:
```diff
@@ -0,0 +1,58 @@
+        vm.errMsg.observe(
+            this,
+            Observer {
+                Toast.makeText(
```
BaseFragment 에 있는 showToast를 사용하시는게 어떠신가요?

BeokBeok replied:
afb139118deb95a813ed0f6e42789b72b6d3f7ba
반영하였습니다.

BeokBeok committed:
HW-6 BaseFragment 의 showToast 함수를 사용하도록 변경

NackUn commented on BeokBeok/app/src/main/java/com/architecturestudy/upbitmarket/UpbitViewModel.kt:
```diff
@@ -10,39 +10,29 @@ class UpbitViewModel(
     init {
-        selectedText.set(MarketTypes.KRW.name)
-        isLoading.set(true)
         upBitRepository.getMarketPrice(
```
init 에서 MarketTypes.KRW.name를 인자로 showMarketPrice 를 호출하는 것은 어떠세요?

BeokBeok replied:
18bf4cf
적용하였습니다.

BeokBeok committed:
HW-6 중복된 구현부 제거

- 리뷰 URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/121/files/afb139118deb95a813ed0f6e42789b72b6d3f7ba#r303721480

BeokBeok committed:
Merge branch 'class02/BeokBeok' of https://github.com/BeokBeok/GoogryAndroidArchitectureStudy into class02/BeokBeok

BeokBeok committed:
HW-6 showMarketPrice 함수가 필요 이상으로 호출되지 않도록 변경

NackUn reviewed (dismissed):
고생하셨습니당!

sjjeong commented on BeokBeok/app/build.gradle:
```diff
@@ -33,7 +33,7 @@ dependencies {
     implementation 'androidx.appcompat:appcompat:1.0.2'
     implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
-    implementation 'com.android.support:design:28.0.0'
+    implementation 'com.android.support:design:29.0.0'
```
com.android.support는 androidx가 아니기 때문에 사용하시면 안돼요.
https://mvnrepository.com/artifact/com.google.android.material/material/1.0.0
이것으로 적용해주세요.

BeokBeok replied:
6e12db737bb06386099898c2c2a7247e62da4884
적용 하였습니다.

sjjeong commented on BeokBeok/app/src/main/java/com/architecturestudy/upbitmarket/UpbitContentsFragment.kt:
```diff
@@ -0,0 +1,58 @@
+import com.architecturestudy.databinding.RvUpbitItemBinding
+
+class UpbitContentsFragment(
+    private val position: Int
```
생성자로 받으면 나중에 재생성 루틴에서 값이 손실 됩니다.
Bundle에 저장해서 가져오는 식으로 변경해주세요.

BeokBeok replied:
a49cbe47b4f70e348726446c75053e41fe6d32dd
적용 하였습니다.

sjjeong commented on BeokBeok/app/src/main/java/com/architecturestudy/upbitmarket/UpbitOutlineFragment.kt:
```diff
@@ -0,0 +1,53 @@
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
```
파괴 주기에서는 super를 나중에 호출하게 해주세요.

BeokBeok replied:
70aee34278a091410f733a4ca81a68e3346be362
적용 하였습니다.

BeokBeok committed:
HW-6 AndroidX 마이그래이션 추가

- Review URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/121#discussion_r304867336

BeokBeok committed:
HW-6 ViewPager 관련 Fragment 생성 시, 생성자가 아닌 Bundle 을 사용하여 값을 전달하도록 변경

- Review URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/121#discussion_r304868194
- Reference URL
https://stackoverflow.com/questions/9245408/best-practice-for-instantiating-a-new-android-fragment

BeokBeok committed:
HW-6 파괴 주기에는 super 를 나중에 호출하도록 변경

- Review URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/121#discussion_r304868779

sjjeong reviewed (approved):
LGTM ✊ ✌️ 👋
